src/com/dotmarketing/startup/runonce/Task03605FixMSSQLMissingConstraints.java

```diff
@@ -0,0 +1,71 @@
+package com.dotmarketing.startup.runonce;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.dotmarketing.startup.AbstractJDBCStartupTask;
+
+/**
+ * This update task will update every <code>VARCHAR</code> and <code>TEXT</code>
+ * column types to <code>NVARCHAR</code> and <code>NVARCHAR(MAX)</code>
+ * respectively.
+ * <p>
+ * Currently, all parameterized DB queries cannot use the indexes on the text
+ * fields because the indexes are for ASCII text. SQL Server JDBC Drivers are
+ * passing query parameters as UTF-8, which forces a whole table scan for all
+ * tables: inode, tree, identifier tables, etc. This makes SQL Server operations
+ * slow. In its current state, it is incapable of serving UTF-8 content.
+ * 
+ * @author Jose Castro
+ * @version 3.7
+ * @since Nov 10, 2016
+ *
+ */
+public class Task03605FixMSSQLMissingConstraints extends AbstractJDBCStartupTask {
```

Comment:
Javadoc updated. In 4.0 there is already a Task named 03710, so making that change would imply renaming lots of upgrade tasks for 4.0 and above.